index.js

```diff
@@ -3,17 +3,31 @@ import createHtmlElement from 'create-html-element';
 // Capture the whole URL in group 1 to keep `String#split()` support
 const urlRegex = () => (/((?<!\+)https?:\/\/(?:www\.)?(?:[-\p{Letter}.]+?[.@][a-zA-Z\d]{2,}|localhost)(?:[-\w\p{Letter}.:%+~#*$!?&/=@]*?(?:,(?!\s))*?)*)/gu);
 
+const parseValue = (value, href) => {
+	switch (typeof value) {
+		case 'function': {
+			return {html: value(href)};
+		}
+
+		case 'undefined': {
+			return {text: href};
+		}
+
+		default: {
+			return {html: value};
+		}
+	}
+};
+
 // Get `<a>` element as string
 const linkify = (href, options = {}) => createHtmlElement({
 	name: 'a',
 	attributes: {
-		href: '',
+		href, // Set first so it appears first in the final HTML string
 		...options.attributes,
-		href, // eslint-disable-line no-dupe-keys
+		href, // eslint-disable-line no-dupe-keys -- Ensures it's not overwritten
```

Comment:
I blamed this to https://github.com/sindresorhus/linkify-urls/commit/72d8831802e9f53c37e0e4e26aabefbe567fea29

I tried changing it and it broke a snapshot just for the order of the attributes. I'm not sure if that was your original intent but I like it.